[release-1.3] Enable OpenShift (#4905)
istio/istio.io: content/en/docs/setup/platform-setup/openshift/index.md

keywords: [platform-setup,openshift]
---

{{< warning >}}

> OpenShift uses `nftables`, which is incompatible with the Istio proxy-init container. Please use [CNI](/docs/setup/additional-setup/cni/) instead.

Reviewer: The nftables support has not been merged. Istio CNI as shipping in Istio 1.3.0 will use iptables.

Reviewer: Istio CNI is using the `iptables` command installed on the worker node, so it may have been using the `nftables`'s `iptables` compatibility layer that may have been installed on your worker nodes. Is this guaranteed to be available on every OpenShift node?

Author: This is 4.1 and above. Clarifying the statement. On those systems, the `iptables` in proxy-init fails (see: https://github.com/istio/istio/issues/13986). In the case of CNI, it uses the host's `iptables` command which is the `nftables` compatibility layer you referenced.

Author: Thanks

> OpenShift uses `nftables`, which is incompatible with the Istio proxy-init container. Please use [CNI](/docs/setup/additional-setup/cni/) instead.

Reviewer: ```suggestion
OpenShift uses `nftables`, which is incompatible with the Istio `proxy-init` container. Please use [CNI](/docs/setup/additional-setup/cni/) instead.
```

keywords: [platform-setup,openshift]
---

{{< warning >}}
OpenShift 4.1 and above use `nftables`, which is incompatible with the Istio `proxy-init` container. Please use [CNI](/docs/setup/additional-setup/cni/) instead.

> >>>>>>> Stashed changes

Reviewer 2: Conflict?

$ oc adm policy add-scc-to-user privileged -z default -n <target-namespace>
{{< /text >}}

Install Istio using either the [Quick Start](/docs/setup/install/kubernetes/) or [Customizable Helm Install instructions](/docs/setup/install/helm/)

After installation is complete, expose an OpenShift route for the ingress gateway.

{{< text bash >}}
$ oc expose svc/istio-ingressgateway --port=80

> {{/text}}

Reviewer 2: ```suggestion
{{< /text >}}
```

        apiVersion: apiserver.config.k8s.io/v1alpha1
        kubeConfigFile: /dev/null
        kind: WebhookAdmission

{{< /text >}}

In the same directory, execute:

{{< text bash >}}

> 

Reviewer 2: Remove leading and trailing blank lines inside `text bash` sections.

Author: Thanks @frankbu. Getting some weird linter issues with this PR and that's where the blank lines are coming from. If I remove the blank lines, the linter complains that I need to have a blank line before and after the tag and mentions a whole bunch of spelling errors for the oc commands. If I add the blank lines in, I get different errors.

Author: Problem was {{/text}} should have been {{< /text >}}

$ cp -p master-config.yaml master-config.yaml.prepatch
$ oc ex config patch master-config.yaml.prepatch -p "$(cat master-config.patch)" > master-config.yaml
$ master-restart api
$ master-restart controllers

{{< /text >}}

## Bookinfo

> Bookinfo uses specifies a user ID of 1337, which is not allowed by default in OpenShift. To allow this user ID to be used, execute the following, substituting `-n default` for the appropriate namespace.

Reviewer 2: ```suggestion
Bookinfo specifies a user ID of 1337, which is not allowed by default in OpenShift. To allow this user ID to be used, execute the following commands. Replace`-n bookinfo` with the appropriate namespace.
```

$ master-restart api
$ master-restart controllers
{{< /text >}}

## Bookinfo

> The Bookinfo application specifies a user ID of 1337, which is not allowed by default in OpenShift. To allow this user ID to be used, execute the following commands. Replace `-n bookinfo` with the appropriate namespace.

Reviewer: This is very surprising. UID 1337 is the one running Envoy in the `istio-proxy` sidecar container. Anything running with that UID bypasses the sidecar proxy, so Bookinfo certainly doesn't run with that UID.
So you probably meant something like:
```suggestion
The Istio sidecar injected into each pod runs with user ID 1337, which is not allowed by default in OpenShift. To allow this user ID to be used, execute the following commands. Replace `-n bookinfo` with the appropriate namespace.
```

Author: Thanks.
